src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -126,6 +130,370 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Theory]
+        [InlineData("normal.zip")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for full framework.")]
+        public static async Task ZipArchiveEntry_CopyTo_CorruptedFile(string zipname)
+        {
+            string filename = zfile(zipname);
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(filename);
+            string tamperedFileName = "binary.wmv";
+
+            int nameOffset = PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 8);  // patch uncompressed size in file header
+            PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 22, nameOffset + tamperedFileName.Length); // patch in central directory too
```

Comment:
A lot of this code looks repeated between the tests, and I can't tell what's actually being tested (and what the difference is).  Maybe it's just a lack of familiarity with the area, but I would really like to better understand what is different between the tests.

Reply:
Isn't test names not expressing well what they are testing? Will try to improve naming. 
There are tests for testing:
1. if uncompressed size in local header didn't match with central directory record => throwing
2. if uncompressed size in local header didn't match with central directory record => throwing for **64 bit** zip archives
3. if uncompressed size in data descriptor didn't match with central directory record => throwing for **unseekable** zip file
4. In **Read mode** test Stream.Read(...) is limiting output up to uncompressed size
5. In **Read mode** test Stream.CopyTo(...)/CopyToAsync() is limiting output up to uncompressed size
6. In **Read mode** test Stream.Read(...) is limiting output up to uncompressed size - for **64 bit archives**
7. In **Read mode** test Stream.CopyTo(...)/CopyToAsync() is limiting output up to uncompressed size - for **64 bit archives**
8. In **Update** mode test append to corrupted file works as expected
9. In **Update** mode test add a file to zip archive having corrupted file works
10. In **Update** mode test override corrupted file within a zip archive works